src/cocoa/toga_cocoa/widgets/table.py

```diff
@@ -7,27 +7,27 @@ class TogaTable(NSTableView):
     # TableDataSource methods
     @objc_method
     def numberOfRowsInTableView_(self, table) -> int:
-        return len(self.interface.data)
+        return len(self.interface.data.data) if self.interface.data else 0
 
     @objc_method
     def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
         column_index = int(column.identifier)
-        return self.interface.data[row][column_index]
+        return self.interface.data.data[row].data[column_index]
```

Comment:
This indicates a problem with the interface - there's a few too many datas here :-)

This should be behind an API like `self.interface.data.item(row, column_index)`. The data attribute of a ListDataSource shouldn't be exposed like this.